hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -140,6 +144,12 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
 
   protected val partitionColumns: Array[String] = tableConfig.getPartitionFields.orElse(Array.empty)
 
+  /**
+   * if true, need to deal with schema for creating file reader.
+   */
+  protected val dropPartitionColumnsWhenWrite: Boolean =
+    metaClient.getTableConfig.getDropPartitionColumnsWhenWrite && partitionColumns.nonEmpty
```

Comment:
call partitionColumns.nonEmpty first to shortcut condition check ?

Reply:
There are very few cases in which  enable `hoodie.datasource.write.drop.partition.columns`.
i think the current logical will work better to shortcut condition check.